### ISSUE TITLE
Make HttpMethodActionConstraint public in Core 3.0

### ISSUE THREAD

**Issue by EvilBeaver** [aspnet/AspNetCore#8653]
### Is your feature request related to a problem? Please describe.
I'm making my own ApplicationModelProvider. Before Core 3.0 I used an HttpMethodActionConstraint in my ActionModels to filter out actions by http method.

In Core 3.0 HttpMethodActionConstraint was made an internal class. Why?

Example: 'm trying to do

```c#
var selectorModel = new SelectorModel();
selectorModel.ActionConstraints.Add(new HttpMethodActionConstraint(httpMethods));
```

### Describe the solution you'd like
I want to use an existing reusable class ``HttpMethodActionConstraint`` from ``Microsoft.AspNetCore.Mvc.ActionConstraints`` instead of making my own.

### Describe alternatives you've considered
alternative solution. Copy/Paste source of ``Microsoft.AspNetCore.Mvc.ActionConstraints.HttpMethodActionConstraint`` in my project

### Additional context
Custom ApplicationModelProvider is used for integration with database-provided model. I don't need an AspNetCore's class-based model, but want to use all other stuff. AppModels are great and limitation of their's building blocks is making me sad.

**Comment by pranavkm** [comment]
The type was `pubinternal` in 2.x and we defaulted to making it internal as part of #4932. If your app uses endpoint-routing, you may use [`HttpMethodMetadata`](https://github.com/aspnet/AspNetCore/blob/c565386a3ed135560bc2e9017aa54a950b4e35dd/src/Http/Routing/src/HttpMethodMetadata.cs) instead. It's what MVC does under the hood - https://github.com/aspnet/AspNetCore/blob/c565386a3ed135560bc2e9017aa54a950b4e35dd/src/Mvc/Mvc.Core/src/Routing/ActionEndpointFactory.cs#L249-L253.

If you are unable to use endpoint routing, we'd be happy to consider a PR to make the type public.

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us. We believe that the question you've raised have been answered. If you still feel a need to continue the discussion, feel free to reopen it and add your comments.

**Comment by EvilBeaver** [comment]
Yes, thank you for the answer.

**Comment by ismcagdas** [comment]
Hi @pranavkm sorry for writing to a closed issue but we have a similar problem. We are using HttpMethodActionConstraint in two places of our custom IApplicationModelConvention implementation to prevent usage of http verbs for some auto generated controllers, see below lines:

https://github.com/aspnetboilerplate/aspnetboilerplate/blob/dev/src/Abp.AspNetCore/AspNetCore/Mvc/Conventions/AbpAppServiceConvention.cs#L122
https://github.com/aspnetboilerplate/aspnetboilerplate/blob/dev/src/Abp.AspNetCore/AspNetCore/Mvc/Conventions/AbpAppServiceConvention.cs#L229

I tried to change this but couldn't succeed without using HttpMethodActionConstraint. Do you think it is possible to achieve this without using HttpMethodActionConstraint ?

**Comment by rynowak** [comment]
We've heard enough feedback about this particular issue that we're going to make this public.

**Comment by rynowak** [comment]
Along with `AmbiguousActionException`

**Comment by rynowak** [comment]
Fixed by: #11969
